File: Java/OperatingSystemClasses.java
//package EnumerationOperatingSystem;

public enum OperatingSystemClasses {

    //CONTENT
    DESKTOP(1),
    SERVER(2),
    EMBEDDED(3);
    //CONTENT

    private int numericValue;

    OperatingSystemClasses(int numericValue){
        this.numericValue = numericValue;
    }

    public OperatingSystemClasses parse(String stringToParse){
        for (OperatingSystemClasses oSC:OperatingSystemClasses.values())
        {
            if (stringToParse.toLowerCase().contains(oSC.name().toLowerCase()))
            {
            	return oSC;
            }
        }
        return OperatingSystemClasses.DESKTOP;
    
    }

    public int GetNumericValue(){
        return this.numericValue;
    }
}

File: Python/OperatingSystemClasses.py
from enum import Enum
import re

class OperatingSystemClasses(Enum):
    	
	#ENUM:START
	DESKTOP = 1
	SERVER = 2
	EMBEDDED = 3
	#ENUM:END

	def parse(self, string):
		for j in OperatingSystemClasses:
			if (string.lower()).find(j.name.lower())>=0:
				return j
		return OperatingSystemClasses.DESKTOP

